pyomo/contrib/mindtpy/mip_solve.py

```diff
@@ -127,22 +130,39 @@ def setup_mip_solver(solve_data, config, regularization_problem):
     if regularization_problem:
         mainopt = SolverFactory(config.mip_regularization_solver)
     else:
-        mainopt = SolverFactory(config.mip_solver)
+        if config.mip_solver == 'gurobi_persistent' and config.single_tree:
+            @SolverFactory.register('gurobi_persistent4mindtpy', doc='Persistent python interface to Gurobi')
```

Comment:
Why are you registering this (temporary) solver with the SolverFactory? Do you ever see a case where a user would want to create another instance of this solver outside of mindpy?

I would just create an instance of this new solver and return it (and avoid the Factory)

Reply:
This solver will only be used in MindtPy. I thought this solver needs first to be registered for later use even just in MindtPy. Is there an example that only create a instance of the solver？

Reply:
All you need to do is make an instance of the class (that is all the SolverFactory does).  In this case, I think something like the following would work:
```python
mainopt = GurobiPersistent4MindtPy()
mainopt.solve_data = solve_data
mainopt.config = config
```
I also think that the solver can be declared at the module scope ... and doesn't have to be a "temporary" class declared in the method.